src/pages/transactions/[id].js

```diff
@@ -1,14 +1,24 @@
 import { useRouter } from "next/router";
+import { useState } from "react";
 import transactions from '../../data/transactions.json';
 import Link from "next/link";
 import ChevronLeft from "@/src/assets/icons/chevron_left.svg";
+import DeleteBin from "src/assets/icons/delete.svg"
 
-export default function TransactionDetails() {
+export default function TransactionDetails({handleDelete}) {
   const router = useRouter();
   const { id } = router.query;
+  const [modal, setModal] = useState(false);
```

Comment:
If you plan to implement more states e.g. edit, delete, update - it might be easier to use one State and define different modus for the modal e.g. default modus is empty and changed depend on the functionality e.g. isEdit, isDelete, isUpdate.